export class AppError extends Error {
    statusCode: number;
    
    constructor(message: string, statusCode: number) {
      super(message);
      this.statusCode = statusCode;
      this.name = this.constructor.name;
    }
  }
  
  export class BadRequestError extends AppError {
    constructor(message = "Solicitud incorrecta") {
      super(message, 400);
    }
  }
  
  export class UnauthorizedError extends AppError {
    constructor(message = "No autorizado") {
      super(message, 401);
    }
  }
  
  export class ForbiddenError extends AppError {
    constructor(message = "Acceso prohibido") {
      super(message, 403);
    }
  }
  
  export class NotFoundError extends AppError {
    constructor(message = "Recurso no encontrado") {
      super(message, 404);
    }
  }
  
  export class ConflictError extends AppError {
    constructor(message = "Conflicto con datos existentes") {
      super(message, 409);
    }
  }
  
  export class InternalServerError extends AppError {
    constructor(message = "Error interno del servidor") {
      super(message, 500);
    }
  }
  